pkg/graveler/committed/meta_range.go

```diff
@@ -24,6 +24,39 @@ type Iterator interface {
 	Close()
 }
 
+// DiffIterator iterates over all Range headers and values of a Diff, allowing seeking by entire
+// ranges.
+type DiffIterator interface {
+	// Next moves to look at the next value in the current Range, or a header for the next Range if the current Range is over and a next range exists.
+	Next() bool
+	// NextRange skips the current range
+	// possible only if we are currently inside a range
+	// the next value could be Range or Value
+	NextRange() bool
+	// Value returns a nil ValueRecord and a Range before starting a Range, or a Value and that Range when inside a Range.
+	// In contrast to Iterator, the DiffIterator might not have a current range -  this could happen if the current value exists in two different ranges
```

Comment:
Can we explain what `NextRange` does in this case?

Reply:
@arielshaqed , I changed the documentation of the `DiffIterator`, PTAL